src/js/clock.js

```diff
@@ -1,27 +1,43 @@
 window.App = window.App || {};
 
 /**
- * Simple and efficient clock.
+ * Simple and efficient clock with 12h/24h format support.
  */
 App.Clock = function () {
     this.time = '';
     this.clockElement = document.getElementById('clock');
+    this.format = localStorage.getItem('clockFormat') || '24h';
 
     this.update();
 };
 
+// Set new format dynamically
+App.Clock.prototype.updateFormat = function (newFormat) {
+    this.format = newFormat;
+    this.time = ''; // Reset to force immediate update
+    this.setTime();
+};
+
 App.Clock.prototype.setTime = function () {
     const today = new Date();
-
-    const h = today.getHours();
-    // Add zero in front of numbers < 10
+    let h = today.getHours();
     const m = ('0' + today.getMinutes()).slice(-2);
+    let period = ''; // Used for 12-hour format
+
+    if (this.format === '12h') {
+        // Convert to 12-hour format
+        period = h >= 12 ? 'PM' : 'AM';
+        h = h % 12 || 12; // Convert "0" hour to "12"
+    }
 
-    const currentTime = h + ':' + m;
+    const currentTime =
+        this.format === '24h'
+            ? `${h}:${m}`
+            : `${h}:${m}<span class="clock-period">${period}</span>`;
 
     if (currentTime !== this.time) {
         this.time = currentTime;
-        this.clockElement.innerText = currentTime;
+        this.clockElement.innerHTML = currentTime;
```

Comment:
We must stick to using `innerText`. The `innerHTML` usage was the reason why Firefox rejected the extension a while ago: https://github.com/superKalo/crypto-tab/pull/47

Reply:
MB, I forgot about that. I've reworked the logic to use innerText instead of innerHTML.